doc/api_module.rst

```diff
@@ -0,0 +1,230 @@
+
+    Example::
+
+        #!/usr/bin/python3
+        import dnf
+
+        base = dnf.Base()
+        base.read_all_repos()
+        base.fill_sack()
+
+        module_base = dnf.module.module_base.ModuleBase(base)
+        module_packages, nsvcap = module_base.get_modules('nodejs:11/minimal')
+
+        print("Parsed NSVCAP:")
+        print("name:", nsvcap.name)
+        print("stream:", nsvcap.stream)
+        print("version:", nsvcap.version)
+        print("context:", nsvcap.context)
+        print("arch:", nsvcap.arch)
+        print("profile:", nsvcap.profile)
+
+        print("Matching modules:")
+        for mpkg in module_packages:
+            print(mpkg.getFullIdentifier())
+
+
+
+
+.. class:: libdnf.module.ModulePackage
+
+This class represents a record identified by NSVCA from the repository modular metadata. See also https://github.com/fedora-modularity/libmodulemd/blob/master/spec.v2.yaml.
+
+  .. method:: getName()
+
+    Return name of the module.
+
+  .. method:: getStream()
+
+    Return stream of the module.
+
+  .. method:: getVersion()
+
+    Return version of the module as a string.
+
+  .. method:: getVersionNum()
+
+    Return version of the module as a number.
+
+  .. method:: getContext()
+
+    Return context of the module.
+
+  .. method:: getArch()
+
+    Return architecture of the module.
+
+  .. method:: getNameStream()
+
+    Return string in the form of 'name:stream' for the module.
+
+  .. method:: getNameStreamVersion()
+
+    Return string in the form of 'name:stream:version' for the module.
+
+  .. method:: getFullIdentifier()
+
+    Return string in the form of 'name:stream:version:context:architecture' for the module.
+
+  .. method:: getProfiles(name=None)
+
+    Return tuple of :class:`libdnf.module.ModuleProfile` objects representing each of the individual profiles of the module. If the `name` is given, only profiles matching the `name` pattern are returned.
+
+  .. method:: getSummary()
+
+    Return summary of the module.
+
+  .. method:: getDescription()
+
+    Return description of the module.
+
+  .. method:: getRepoID()
+
+    Return identifier of source repository of the module.
+
+  .. method:: getArtifacts()
+
+    Return tuple of the artifacts of the module.
+
+  .. method:: getModuleDependencies()
+
+    Return tuple of :class:`libdnf.module.ModuleDependencies` objects representing modular dependencies of the module.
+
+  .. method:: getYaml()
+
+    Return repomd yaml representing the module.
+
+  .. method:: addStreamConflict(package)
+
+    Add conflict with a module stream `package` represented as a ModulePackage.
+
+  .. method:: getId()
+
+  .. method:: getContextCStr()
+
+  .. method:: getArchCStr()
+
+  .. method:: getNameCStr()
+
+  .. method:: getStreamCStr()
```

Comment:
@j-mracek I'm thinking about dropping these methods from API documentation. What to you think?

Reply:
I do support dropping it.